src/schedule/schedule_dataflow_rewrite.cc

```diff
@@ -756,12 +756,19 @@ Array<Tensor> Schedule::rfactor(const Tensor& tensor,
   VarReplacer replacer(vsub);
   Array<Expr> new_source = ir::UpdateArray(reduce->source,
     [&replacer] (const Expr& e) { return replacer.Mutate(e); });
+
+  Array<Expr> orig_pred;
+  orig_pred.push_back(predicate);
+
+  Array<Expr> new_pred = ir::UpdateArray(orig_pred,
+    [&replacer] (const Expr& e) { return replacer.Mutate(e); });
+
```

Comment:
Why do you create an array?
Just use
`Expr new_pred = replacer.Mutate(predicate)`

Reply:
Thank you for the syntax, updated.